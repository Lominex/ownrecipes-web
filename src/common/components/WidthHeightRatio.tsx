import classNames from 'classnames';

import '../css/width_height_ratio.css';

export interface IWidthHeightRatioProps {
  height: number;
  width:  number;

  className?: string;
  children: React.ReactNode;
}

const WidthHeightRatio: React.FC<IWidthHeightRatioProps> = ({
  height, width, className, children }: IWidthHeightRatioProps) => (
    <div
        style = {{
          paddingTop: `${height}%`,
          width:      `${width}%`,
          position:   'relative',
        }}
        className={classNames('width-height-ratio', className)}>
      {children}
    </div>
);

export default WidthHeightRatio;
