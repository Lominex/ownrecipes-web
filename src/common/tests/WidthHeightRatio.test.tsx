import renderer from 'react-test-renderer';

import WidthHeightRatio from '../components/WidthHeightRatio';

test('WidthHeightRatio', () => {
  const component = renderer.create(
    <WidthHeightRatio
        height={50}
        width={75}
        className='some class'
        >
      Some string test
    </WidthHeightRatio>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
