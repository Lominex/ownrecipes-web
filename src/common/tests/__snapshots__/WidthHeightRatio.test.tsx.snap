// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidthHeightRatio 1`] = `
<div
  className="width-height-ratio some class"
  style={
    Object {
      "paddingTop": "50%",
      "position": "relative",
      "width": "75%",
    }
  }
>
  Some string test
</div>
`;
